source/hpack.c

```diff
@@ -86,7 +91,8 @@ int aws_hpack_decode_integer(struct aws_byte_cursor *to_decode, uint8_t prefix_s
 
     uint8_t byte = 0;
     if (!aws_byte_cursor_read_u8(to_decode, &byte)) {
-        return AWS_OP_ERR;
+        AWS_FATAL_ASSERT(false); /* Like 5 seconds ago we made sure there was data in here */
+        return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
```

Comment:
Just a regular assert is fine for what would be a programming error. regular assert is compiled out. Also you've still got some error handling left over here
```
if (!read())
{
    assert(0); /* like 4 seconds ago */
}
```

or just don't bother checking that read succeeded here at all